### ISSUE TITLE
[PROPOSAL] Facet for file format type

### ISSUE THREAD

**Issue by dolfinus** [OpenLineage/OpenLineage#2770]
**Purpose:**

Spark integration with OL can collect information about dataset location (warehouse+metastore) and its schema. But file format is currently collected only for Iceberg implementation, non-Iceberg tables have no such information. This can be important for data discovery.

**Proposed implementation**

Currently there is a facet [StorageDatasetFacet](https://github.com/OpenLineage/OpenLineage/blob/b107e409bbe526ba491bb5bdb73a6036d80cf800/spec/facets/StorageDatasetFacet.json) with fields `storageLayer` and `fileFormat`. But `fileFormat` is optional and `storageLayer` is not, which does not make sense for non-Iceberg tables.

I see 3 options here:
* Update `StorageDatasetFacet` spec to name `storageLayer` optional, and `fileFormat` mandatory. But this may conflict with facet name.
* Passing some static value to `storageLayer` field, e.g. `native` or `none`, without any changes in `StorageDatasetFacet` sper.
* Adding another facet just for `fileFormat`. But having 2 slightly different facets is cumbersome.

**Comment by mobuchowski** [comment]
Why not just add file format where it makes sense using existing type?

`storageLayer` should indicate what the storage is anyway - pure format is sometimes not enough to reason how to interact with the data anyway

**Comment by dolfinus** [comment]
> Why not just add file format where it makes sense using existing type?

Do you mean option 2?

**Comment by mobuchowski** [comment]
>Passing some static value to storageLayer field, e.g. native or none, without any changes in StorageDatasetFacet sper.

Basically, yeah - but providing information about what `native` means - local filesystem? Object storage?

I think the constraints in the facet are needlessly narrow:
```
"Storage layer provider with allowed values: iceberg, delta."
"File format with allowed values: parquet, orc, avro, json, csv, text, xml."
```

**Comment by mobuchowski** [comment]
Yeah, I agree - we should encourage some standarization though, even if people can actually put everything there and event still be valid with regards to schema, similar to what we do with dataset naming.

**Comment by dolfinus** [comment]
> but providing information about what native means - local filesystem? Object storage?

I struggle to find a value which is both meaningful for users and does follow the original `storageLayer` definition. `spark`? `hive`? `default`?
